Makefile

```diff
@@ -92,27 +92,27 @@ depend: depend.ensure
 ${GOPATH}/bin/dep:
 	go get -u github.com/golang/dep/cmd/dep
 
-Gopkg.lock: Gopkg.toml ; $(info $(H) generating) @
-	$(Q) dep ensure -update
+Gopkg.lock: Gopkg.toml | ${GOPATH}/bin/dep ; $(info $(H) generating) @
+	$(Q) ${GOPATH}/bin/dep ensure -update
 
```

Comment:
I think it is possible your $PATH does not include $GOPATH/bin as documented here: https://github.com/istio/istio/blob/master/DEV-GUIDE.md#setting-up-environment-variables

Reply:
You're probably right, and this does help to explain how/why someone came up with ${GOPATH}/bin as a location for squirreling away bits.  My change seems to get rid of this requirement, though I can undo it if people really want people to be modifying their paths like this.  I was brought up on the philosophy that a build should be able to work even with an empty path (though I have been turning a blind eye to dependencies on go and other common tools).

Reply:
The PATH setting is in DEV-GUIDE.md, and I think it should stay there for principle of least surprise.  I don't feel strongly about this particular problem.  The code is a little harder to read as a downside.

Maybe a different reviewer feels strongly.

Overall really nice work 👍

Reply:
The "|" makes anything that follows be an order-only dependency, which basically means it needs to be run first but the relative age of the dependency is irrelevant about whether to run the step.  In this situation I was trying to declare that "dep" should exist but if "dep" gets updated then this shouldn't necessary cause the recipe to be re-run.  Currently it probably only adds value for people who opt to update "dep" on their own.  I thought it'd also be useful if the rule for "dep" was made an always run (so it more aggressively tried to pull in the latest if some runs "make dep"), though I just tried to make a mock test on my Mac and it still wanted to update "dep" even though it was order only.  If the | is too obscure then I can remove it.  I was mostly trying to be conservative in the dependency that I was adding.